### PR TITLE
[feature]icm42670p driver support

### PR DESCRIPTION
**Pull request by ShiauweiZhao**

### CODE REVIEW

File: src/drivers/imu/invensense/icm42670p/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,46 @@
+############################################################################
+#
+#   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/drivers/imu/invensense/icm42670p/CMakeLists.txt:3]:
```suggestion
#   Copyright (c) 2021 PX4 Development Team. All rights reserved.
```

File: src/drivers/imu/invensense/icm42670p/ICM42670P.hpp [hunk 2]
```diff
@@ -0,0 +1,184 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/drivers/imu/invensense/icm42670p/ICM42670P.hpp:3]:
```suggestion
 *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
```

File: src/drivers/imu/invensense/icm42670p/InvenSense_ICM42670P_registers.hpp [hunk 3]
```diff
@@ -0,0 +1,293 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2020 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/drivers/imu/invensense/icm42670p/InvenSense_ICM42670P_registers.hpp:3]:
```suggestion
 *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
```

File: src/drivers/imu/invensense/CMakeLists.txt [hunk 4]
```diff
@@ -41,3 +41,4 @@ add_subdirectory(icm42688p)
 add_subdirectory(mpu6000)
 add_subdirectory(mpu6500)
 add_subdirectory(mpu9250)
+add_subdirectory(icm42670p)
```
Comment (dagar) [inline src/drivers/imu/invensense/CMakeLists.txt:44]: Can you keep this list sorted?

File: src/drivers/drv_sensor.h [hunk 5]
```diff
@@ -108,7 +108,7 @@
 #define DRV_DIFF_PRESS_DEVTYPE_SDP33    0x4B
 #define DRV_BARO_DEVTYPE_LPS33HW        0x4C
 #define DRV_BARO_DEVTYPE_TCBP001TA      0x4D
-
+#define DRV_IMU_DEVTYPE_ICM42670P       0x4E
```
Comment (dagar) [inline src/drivers/drv_sensor.h:111]: Maybe move it up to 0x2A which is grouped with more IMUs including the DRV_IMU_DEVTYPE_ICM42605?

**Comment by dagar** [comment]
Looks good, has this had any testing on hardware?

**Comment by ShiauweiZhao** [comment]
image
image
image
the error_count is fifo_empty becasue i use fifo read imu data 
image

**Comment by dagar** [comment]
Taking a look at the change I see that you have the output data rate at 800 Hz and the filters set. What we've settled on for the the other drivers is to disable all sensor side processing and get full raw data (or as fast as we possibly need it) via the FIFO and then perform all filtering within PX4. This way you can see what's really happening with the system and the filtering is completely user configurable.

The other setting that's a bit low is the gyro is the gyro range (250 degrees/second).

Would you mind if we sync up the configuration of the icm42670p to be roughly in line with the icm42650 (or similar)?

**Review by dagar — APPROVED** [review]

**Pull request merged**